roachpb/data.proto

```diff
@@ -66,22 +66,25 @@ enum ValueType {
   TIMESERIES = 100;
 }
 
-// Value specifies the value at a key. Multiple values at the same key
-// are supported based on timestamp.
+// Value specifies the value at a key. Multiple values at the same key are
+// supported based on timestamp. The data stored within a value is typed
+// (ValueType) and custom encoded into the raw_bytes field. A custom encoding
+// is used instead of separate proto fields to avoid proto overhead and to
+// avoid unnecessary encoding and decoding as the value gets read from disk and
+// passed through the network. The format is:
+//
+//   <4-byte-checksum><1-byte-tag><encoded-data>
+//
+// A CRC-32-IEEE checksum is computed from the associated key, tag and encoded
+// data, in that order.
+//
+// TODO(peter): Is a 4-byte checksum overkill when most (all?) values
+// will be less than 64KB?
```

Comment:
Just my estimate of what a likely upper bound on a typical column / row size will be.